### PR TITLE
Ticket #10388

### PR DESCRIPTION
**Pull request by jremes-foss**
As per @jksuom mentioned:

``` python
if arg.is_Number:
```

Replaced by -->

``` python
if arg.is_Integer:
```

This should solve ticket #10388. Thank you.

### CODE REVIEW

File: sympy/functions/combinatorial/factorials.py [hunk 1]
```diff
@@ -255,7 +255,7 @@ def _eval(self, n):
 
     @classmethod
     def eval(cls, arg):
-        if arg.is_Number:
```
Comment (jksuom) [inline sympy/functions/combinatorial/factorials.py]: This line is in ``subfactorial``. The issue was about ``factorial2``.

Comment (Newman101) [inline sympy/functions/combinatorial/factorials.py]: Whoops, sorry - trying to get used to emacs. I did not notice that one. I'll re-factor this then.

**Comment by gxyd** [comment]
@Newman101 
In this case `.is_Integer` should be used not `.is_integer`.
Let me point the difference: `.is_Integer` means that the object is an instance of `Integer` class (from SymPy),    while `.is_integer` checks for whether the assumption of `integer=True` is there or not.

Consider this:
```
>>> x = Symbol('x', integer=True)
>>> x.is_Integer
False          # x is an instance of Symbol class, not of Integer
>>> x.is_integer
True           # x has assumption of integer=True
```

so change `.is_integer` -> `.is_Integer`.
You also need to fix the merge conflicts, you can [read this](https://github.com/sympy/sympy/wiki/Development-workflow#id48)

**Comment by gxyd** [comment]
Closing this, since the issue has been handled by PR #10478 .

**Pull request closed without merging**